src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -1209,6 +1227,16 @@ void IMergeTreeDataPart::loadRowsCount()
             return;
         }
 
+        for (const NameAndTypePair & column : columns)
+        {
+            if (column.name == LightweightDeleteDescription::FILTER_COLUMN.name)
+            {
+                ColumnPtr column_col = column.type->createColumn(*getSerialization(column.name));
+                masked_rows_count = column_col->getNumberOfDefaultRows();
```

Comment:
TODO:
This is the backwards compatibility implementation -- for a part without this metadata, we calculate the count manually.
That involves scanning the _row_exists column, which is somewhat expensive.
But in essence that is also what happens with `select count()` currently anyway, so maybe this is acceptable?